### PR TITLE
[Analyzer] BinaryEdge

### LINKED ISSUE #1259: [Analyzer] BinaryEdge
## Name
BinaryEdge

## Link
https://www.binaryedge.io/

## Type of analyzer
IP: https://docs.binaryedge.io/api-v2/#v2queryiptarget and https://docs.binaryedge.io/api-v2/#v2querysearch
Domains: several, starting from https://docs.binaryedge.io/api-v2/#v2querydomainssubdomaintarget

## Why should we use it
Free API up to 250 requests at month

### PR DESCRIPTION
**Pull request by shivam-Purohit**
Added BinaryEdge API that scans for IP and domain closes #1259
# Description

BinaryEdge analyzer scan against IP and domain fixes [issue](https://github.com/intelowlproject/IntelOwl/issues/1259) 

## Type of change

- [x] New feature (non-breaking change which adds functionality).

# Checklist

- [x] I have read and understood the rules about [how to Contribute](https://intelowl.readthedocs.io/en/latest/Contribute.html) to this project
- [x] The pull request is for the branch `develop`
- [x] A new plugin (analyzer, connector or playbook) was added or changed, in which case:
    - [x] [Usage](https://github.com/intelowlproject/IntelOwl/blob/master/docs/source/Usage.md) file was updated.
    - [ ] [Advanced-Usage](./Advanced-Usage.md) was updated (in case the plugin provides additional optional configuration).
    - [x] Secrets were added in [env_file_app_template](https://github.com/intelowlproject/IntelOwl/blob/master/docker/env_file_app_template), [env_file_app_ci](https://github.com/certego/IntelOwl/blob/master/docker/env_file_app_ci) and in the [Installation](./Installation.md) docs, if necessary.
    - [x] If the plugin requires mocked testing, `_monkeypatch()` was used in its class to apply the necessary decorators.
    - [ ] If a File analyzer was added, its name was explicitly defined in [test_file_scripts.py](https://github.com/intelowlproject/IntelOwl/blob/master/tests/analyzers_manager/test_file_scripts.py) (not required for Observable Analyzers).
    - [ ] If you created a new analyzer and it is free (does not require API keys), please add it in the `FREE_TO_USE_ANALYZERS` playbook in `playbook_config.json`
    - [x] I have provided the resulting raw JSON of a finished analysis and a screenshot of the results.
- [ ] If external libraries/packages with restrictive licenses were used, they were added in the [Legal Notice](https://github.com/certego/IntelOwl/blob/master/.github/legal_notice.md) section.
- [x] Linters (`Black`, `Flake`, `Isort`) gave 0 errors. If you have correctly installed [pre-commit](https://intelowl.readthedocs.io/en/latest/Contribute.html#how-to-start-setup-project-and-development-instance), it does these checks and adjustments on your behalf.
- [ ] I have added tests for the feature/bug I solved. All the tests (new and old ones) gave 0 errors.
- [] If changes were made to an existing model/serializer/view, the docs were updated and regenerated (check [CONTRIBUTE.md](./Contribute.md)).
- [ ] If the GUI has been modified:
    - [ ] I have a provided a screenshot of the result in the PR.
    - [ ] I have created new frontend tests for the new component or updated existing ones.

### Important Rules
- If you miss to compile the Checklist properly, your PR won't be reviewed by the maintainers.
- If your changes decrease the overall tests coverage (you will know after the Codecov CI job is done), you should add the required tests to fix the problem
- Everytime you make changes to the PR and you think the work is done, you should explicitly ask for a review. After being reviewed and received a "change request", you should explicitly ask for a review again once you have made the requested changes.

### CODE REVIEW

**Comment by shivam-Purohit** [comment]
I was trying to test the changes but came across an app-breaking error. It was related to different files that I was not working on. I thought there was a problem with my repo. So I recloned and reinstalled the app but still, the problem is persistent.
I have attached the screenshot of the terminal. 
ubuntu  Running  - Oracle VM VirtualBox 19-02-2023 10_48_16

 ERROR/MainProcess] Signal handler <function worker_ready_connect at 0x7fe4f33f63a0> raised: ValidationError({'OneNote_Info': {'python_module': [ErrorDetail(string="`python_module` incorrect, api_app.analyzers_manager.file_analyzers.onenote.OneNoteInfo couldn't be imported", code='invalid')]}})

**Comment by 0ssigeno** [comment]
The local error is a version missmatch between the code present inside the container and the volume mounted in the container(you have an analyzer_config.json that has the onenote analyzer but inside the container there is no python code for that analyzer).
If you rebuild the containers from scratch you should be able to fix it.

File: api_app/analyzers_manager/observable_analyzers/binaryedge.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+from api_app.analyzers_manager import classes
+from api_app.exceptions import AnalyzerRunException
+from tests.mock_utils import MockResponse, if_mock_connections, patch
+
+
+class BinaryEdge(classes.ObservableAnalyzer):
+    base_url: str = "https://api.binaryedge.io/v2/query/"
+
+    def set_params(self, params):
+        self.__api_key = self._secrets["api_key_name"]
```
Comment (0ssigeno) [inline api_app/analyzers_manager/observable_analyzers/binaryedge.py:15]:
```python3
        if not self.__api_key:
            raise AnalyzerConfigurationException("API key is required")
```

File: configuration/analyzer_config.json [hunk 2]
```diff
@@ -203,6 +203,28 @@
...
+    "leaks_info": false,
+    "observable_supported": ["ip","domain"],
+    "config": {
+      "soft_time_limit": 30,
+      "queue": "default"
+    },
+    "secrets": {
+      "api_key_name": {
+        "env_var_key": "BINARYEDGE_KEY",
+        "description": "",
```
Comment (0ssigeno) [inline configuration/analyzer_config.json:221]: "BinaryEdge api key"

**Review by 0ssigeno — CHANGES_REQUESTED** [review]

**Comment by 0ssigeno** [comment]
If possible add a screenshot of the result, once you are able to fix the local instance

**Comment by shivam-Purohit** [comment]
ubuntu  Running  - Oracle VM VirtualBox 20-02-2023 20_14_25
ubuntu  Running  - Oracle VM VirtualBox 20-02-2023 20_17_15

This is for both domain and IP. @0ssigeno Can you please review?  Suggest changes if needed.

**Comment by 0ssigeno** [comment]
The last missing thing is the dummy env key on `env_file_app_ci`, this will allow the test to run the analyzer, and if the mockup are correctly set, the test will pass.
Once this is done, the PR is mergeable.

**Comment by shivam-Purohit** [comment]
> The last missing thing is the dummy env key on `env_file_app_ci`, this will allow the test to run the analyzer, and if the mockup are correctly set, the test will pass. Once this is done, the PR is mergeable.

is it about the manual tests run on analyzers?
when I run the test against BinaryEdge this is the response I am getting.
Traceback (most recent call last):
  File "/opt/deploy/intel_owl/tests/analyzers_manager/__init__.py", line 75, in test_pipeline
    poll_result = PollingFunction(self)
  File "/opt/deploy/intel_owl/tests/__init__.py", line 111, in PollingFunction
    self.fail("test timed out")
AssertionError: test timed out

----------------------------------------------------------------------
Ran 1 test in 302.331s

**Comment by 0ssigeno** [comment]
The CI now is successful, I am going to merge.
Nice job !

**Pull request merged**